### ISSUE TITLE
Unknown plugin "babel-plugin-transform-object-rest-spread"

### ISSUE THREAD

**Issue by rnmalone** [babel/babel#7207]
Good morning,

I am trying to build my project, but have run into an issue at the last hurdle.

> ERROR in ./client/main.js
Module build failed: ReferenceError: Unknown plugin "babel-plugin-transform-object-rest-spread" specified in "/opt/portal/.babelrc" at 0, attempted to resolve relative to "/opt/portal"

I had to add this plugin for a helper function and it's caused more issues than its solved, though likely my incompetence is at fault.

.babelrc
`{
  "plugins": ["transform-object-rest-spread"]
}`

package.json
`"devDependencies": {
    "babel-plugin-transform-object-rest-spread": "^6.26.0", ... }`

webpack.config.js
`config.module.rules.push(
  {
    test: /\.js$/,
    exclude: /node_modules/,
    use: {
      loader: 'babel-loader',
      options: project.build.babel
    }
  }
);`

I am also mythed as I can't seem to find any other babel config files to explicitly add this plugin?

Any assistance would be appreciated greatly :^)

**Comment by xtuc** [comment]
Your configuration looks correct, just to be sure did you run `npm install`?

**Comment by ccabo1** [comment]
Running into the same issue--works fine in development though not when deployed to heroku:
```
2018-01-13T18:38:47.280671+00:00 app[web.1]: ERROR in ./frontend/index.js
2018-01-13T18:38:47.280673+00:00 app[web.1]: Module build failed: ReferenceError: Unknown plugin "transform-object-rest-spread" specified in "/app/.babelrc" at 1, attempted to resolve relative to "/app"
2018-01-13T18:38:47.280674+00:00 app[web.1]:     at /app/node_modules/babel-core/lib/transformation/file/options/option-manager.js:180:17
2018-01-13T18:38:47.280675+00:00 app[web.1]:     at Array.map (<anonymous>)
2018-01-13T18:38:47.280677+00:00 app[web.1]:     at Function.normalisePlugins (/app/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
2018-01-13T18:38:47.280678+00:00 app[web.1]:     at OptionManager.mergeOptions (/app/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
2018-01-13T18:38:47.280679+00:00 app[web.1]:     at OptionManager.init (/app/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
2018-01-13T18:38:47.280680+00:00 app[web.1]:     at File.initOptions (/app/node_modules/babel-core/lib/transformation/file/index.js:212:65)
2018-01-13T18:38:47.280681+00:00 app[web.1]:     at new File (/app/node_modules/babel-core/lib/transformation/file/index.js:135:24)
2018-01-13T18:38:47.280682+00:00 app[web.1]:     at Pipeline.transform (/app/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
2018-01-13T18:38:47.280683+00:00 app[web.1]:     at transpile (/app/node_modules/babel-loader/lib/index.js:46:20)
2018-01-13T18:38:47.280685+00:00 app[web.1]:     at Object.module.exports (/app/node_modules/babel-loader/lib/index.js:163:20)
```

**Comment by xtuc** [comment]
@ccabo1 could you please share your package.json?

**Comment by barkncats** [comment]
I am rather new to this, am following a video course.

Everything was working fine, then 2 days ago when I attempted to start project with webpack (compile/dev server) I received a similar message:

 Unknown plugin "babel-plugin-transform-object-rest-spread" 

I have been using yarn to add packages to the project. I tried a remove/add, no error messages, but also no success eliminating error message. When I look at the directory in node-modules, the is no /lib directory in the one for "babel-plugin-transform-object-rest-spread" or "babel-plugin-syntax-object-rest-spread".

npm install in another project seemed to work properly, at least there was a /lib directory with an index.js file in it.

Don't know whether issue with yarn or babel addon.

Apologies for any issues my inexperience may create.

**Comment by barkncats** [comment]
My apologies, inexperience definitely involved.

Clearing the yarn cache, deleting /node_modules and running "yarn install" seems to have resolved my problem.

Not sure what caused the problem since everything was working fine the day before when I stopped webpack. Tried deleting directories and running "yarn install" multiple times. Only found out about yarn cache during web search last hour. Also, should have indicated Win10 is my OS.

Again, my apologies for commenting here.
